components/scream/src/physics/p3/p3_ni_conservation_impl.hpp

```diff
@@ -17,7 +17,7 @@ void Functions<S,D>::ni_conservation(const Spack& ni, const Spack& ni_nucleat_te
 {
   const auto sink_ni = (ni2nr_melt_tend + ni_sublim_tend + ni_selfcollect_tend)*dt;
   const auto source_ni = ni + (ni_nucleat_tend+nr2ni_immers_freeze_tend+nc2ni_immers_freeze_tend)*dt;
-  const auto mask = sink_ni > source_ni && context;
+  const auto mask = sink_ni > source_ni && context && sink_ni > 0;
```

Comment:
!= 0

Reply:
See my comment above re: the same code in the F90. Sorry if I'm just confused.

Reply:
@ambrad , this change does appear to be necessary. When I remove it, I get FPEs in all the p3_tests.

Reply:
I think, as stated by Andrew in another comment, that we might have to check with Aaron. A negative sink is a source, but since we have two distinct vars for sink and sources, I would expect both of them to be positive. If we are not "wasting" time to clip them to 0, then I understand, and this check is fine, but if they were not supposed to get negative in the first place, then we should do a `!=0` check (so that a <0 val can still generate errors, and reveal bugs).